### ISSUE TITLE
WiFi keeps reverting to AP mode

### ISSUE THREAD

**Issue by JSa1987** [pihome-shc/PiHomeHVAC#195]
Hello,

I have a Raspberry Pi Zero W that has been running PiHome well for a couple of years, no issue with the WiFi connection.

A couple of days ago I upgraded to AirMax using the image from pihome.eu and a new SD cars. On the first boot the system came up in AP mode as expected, I picked the SSID of my WiFi and entered the password. Instead of connecting to my WiFi it came up again in AP mode. I repeated this 3-4 times and it then finally connected to my WiFi. At that time I didn't pay much attention to this assuming I might had made a typo while typing the password. After this AirMax stayed connected to my WiFi for a couple of days without issues.

As suggested by @twa127 today I flashed a new SD card with MaxAir_1_00_060921. I put this in the Raspberry Pi Zero W, boot it up and it went to AP mode as expected. I tried to select the SSID and entering the password a lot of times but it would always boot back up in AP mode.
At this point I put back in the SD with AirMax that had been working for a couple of days and also this one booted up in AP mode. Again I tried  to select the SSID and entering the password a lot of times but it would always boot back up in AP mode.
I then put back in the SD with PiHome and this connected to the WiFi without issues at the first attempt. I copied the wpa_supplicant.conf file from the PiHome SD to the one with MaxAir_1_00_060921 (via ssh while in AP mode) but this didn't help... it still keeps booting always in AP mode.

In short, with the exemption of a lucky attempt a couple of days ago, I'm unable to get the AirMax image to join my WiFi: it always boots in AP mode.
Any idea why this might be happening? Is there any way to disable the AP mode functionality and just use the wpa_supplicant.conf file configuration?

**Comment by twa127** [comment]
edit index.php and at line 86 enter $no_ap = 0;

I'll investigate the issue

**Comment by twa127** [comment]
Just tried my RPi 3B and it worked fine - booted, then connected to MaxAirHotSpot, browsed to 192.168.50.5, selected my local router, entered password, clicked on 'Set and Restart', lost connection to the hotspot, reconnected to local router, the RPi then had a wifi connection and I was able to logon as normal.

**Comment by twa127** [comment]
to uninstall autohotspot function enter 'sudo bash /var/www/add_on/Autohotspot/autohotspot-setup.sh' and select option 4

**Comment by JSa1987** [comment]
@twa127 I tired to edit index.php adding `$no_ap = 0;` at line 86, but that didn't help. All it did was that when operating as AP it would show the log in page at 192.168.50.5 instead of the menu to select the SSID.

I the run `bash /var/www/add_on/Autohotspot/autohotspot-setup.sh` and selected option 6 (Autohotspot: Force to a Hotspot or Force to Network if SSID in Range) and that solved the issue. It connected to my SSID.

I then tried to reboot and it went back to AP mode. I run again `bash /var/www/add_on/Autohotspot/autohotspot-setup.sh` option 6 and it connected to my SSID.

So seems that at each boot it needs to be forced to connect back to the SSID. 

Looking at autoshotspot-setup.sh my understanding is that all option 6 does is run (line 718):
```
systemctl restart autohotspot.service
```

**Comment by twa127** [comment]
use option 4 to remove the hotspot function altogether, then just config wpa_supplicant.conf as normal

**Comment by twa127** [comment]
also check network_settings table and make sure ap_mode is set to 0

**Comment by JSa1987** [comment]
You are right, it's easier just to uninstall the hotspot function. I did that and it's now working well.
